homeassistant/components/systemmonitor/config_flow.py

```diff
@@ -0,0 +1,162 @@
+                },
+            )
+        }
+    )
+
+
+async def validate_sensor_setup(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate sensor input."""
+    # Standard behavior is to merge the result with the options.
+    # In this case, we want to add a sub-item so we update the options directly.
+    sensors: list[dict[str, Any]] = handler.options.setdefault(SENSOR_DOMAIN, [])
+    sensors.append(user_input)
+    return {}
+
+
+async def validate_import_sensor_setup(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate sensor input."""
+    # Standard behavior is to merge the result with the options.
+    # In this case, we want to add a sub-item so we update the options directly.
+    sensors: list[dict[str, Any]] = handler.options.setdefault(SENSOR_DOMAIN, [])
+    import_processes: list[str] = user_input["processes"]
+    for process in import_processes:
+        sensors.append({CONF_PROCESS: process})
+
+    async_create_issue(
+        handler.parent_handler.hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2024.7.0",
+        is_fixable=False,
+        is_persistent=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "System Monitor",
+        },
+    )
+    return {}
+
+
+async def validate_remove_sensor(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate remove sensor."""
+    removed_indexes: set[str] = set(user_input[CONF_INDEX])
+
+    # Standard behavior is to merge the result with the options.
+    # In this case, we want to remove sub-items so we update the options directly.
+    entity_registry = er.async_get(handler.parent_handler.hass)
+    sensors: list[dict[str, Any]] = []
+    sensor: dict[str, Any]
+    for index, sensor in enumerate(handler.options[SENSOR_DOMAIN]):
+        if str(index) not in removed_indexes:
+            sensors.append(sensor)
+        elif entity_id := entity_registry.async_get_entity_id(
+            SENSOR_DOMAIN, DOMAIN, slugify(f"process_{sensor[CONF_PROCESS]}")
+        ):
+            entity_registry.async_remove(entity_id)
+    handler.options[SENSOR_DOMAIN] = sensors
+    return {}
+
+
+async def get_sensor_setup_schema(handler: SchemaCommonFlowHandler) -> vol.Schema:
+    """Return process sensor setup schema."""
+    hass = handler.parent_handler.hass
+    processes = await hass.async_add_executor_job(get_all_running_processes)
+    return vol.Schema(
+        {
+            vol.Required(CONF_PROCESS): SelectSelector(
+                SelectSelectorConfig(
+                    options=processes,
+                    mode=SelectSelectorMode.DROPDOWN,
+                    sort=True,
```

Comment:
```suggestion
                    sort=True,
                    custom_value=True
```

It would be good to allow the user to specify processes which are not currently running.